File: leetcode-python-go/844. 比较含退格的字符串/main.go
package main

import (
	"fmt"
)

func build(str string) string {
	s := []byte{}
	for i := range str {
		if str[i] != '#' {
			s = append(s, str[i])
		} else if len(s) > 0 {
			s = s[:len(s)-1]
		}
	}
	return string(s)
}

func backspaceCompare(s, t string) bool {
	return build(s) == build(t)
}

func main() {
	s := "ab##"
	t := "c#d#"
	results := backspaceCompare(s, t)
	fmt.Println(results)
}

File: leetcode-python-go/844. 比较含退格的字符串/main.py
s = "ab##"
t = "c#d#"

def backspaceCompare(S: str, T: str) -> bool:
    def build(s: str) -> str:
        ret = list()
        for ch in s:
            if ch != "#":
                ret.append(ch)
            elif ret:
                ret.pop()
        return "".join(ret)
    
    return build(S) == build(T)

print(backspaceCompare(s, t))